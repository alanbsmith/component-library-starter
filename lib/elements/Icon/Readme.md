This is our generic Icon element. You can use it on its own, or extend its styles and put it inside a block.

```jsx { "props": { "style": { "backgroundColor": "#fafafa" } } }
<div>
  <h5>SIZES</h5>
  <span style={{ padding: "5px" }}><Icon name="home" modifiers={['extraSmall']} /></span>
  <span style={{ padding: "5px" }}><Icon name="home" modifiers={['small']} /></span>
  <span style={{ padding: "5px" }}><Icon name="home" /></span>
  <span style={{ padding: "5px" }}><Icon name="home" modifiers={['large']} /></span>

  <h5>COLORS</h5>
  <span style={{ padding: "5px" }}><Icon name="home" /></span>
  <span style={{ padding: "5px" }}><Icon name="home" modifiers={['textLight']} /></span>
  <span style={{ padding: "5px" }}><Icon name="home" modifiers={['chrome000']} /></span>
</div>
```
