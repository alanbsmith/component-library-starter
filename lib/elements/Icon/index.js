import React from 'react';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';
import { rem } from 'polished';
import { applyStyleModifiers } from "styled-components-modifiers";

import {
  fontColors,
} from "../../modifiers";

const customModifiers = {
  extraSmall: (props) => `
    font-size: ${rem(props.theme.dimensions.baseGrid * 1.5)};
  `,
  small: (props) => `
    font-size: ${rem(props.theme.dimensions.baseGrid * 2)};
  `,
  large: (props) => `
    font-size: ${rem(props.theme.dimensions.baseGrid * 3)};
  `,
};

const modifierConfig = {
  ...fontColors,
  ...customModifiers,
};

function UnstyledIcon(props) {
  return <FontAwesome {...props} />;
};


const Icon = styled(UnstyledIcon)`
  color: ${props => props.theme.colors.text};
  font-size: ${props => rem(props.theme.dimensions.baseGrid * 2.5)};
  ${applyStyleModifiers(modifierConfig)};
`;

/** @component */
export default Icon;
