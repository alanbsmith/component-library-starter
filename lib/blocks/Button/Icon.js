import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { Icon as BasicIcon } from '../../elements';

const modifierConfig = {
  left: (props) => `
    padding-right: ${rem(props.theme.dimensions.baseGrid)};
  `,
  right: (props) => `
    padding-left: ${rem(props.theme.dimensions.baseGrid)};
  `,
};

const Icon = styled(BasicIcon)`
  font-size: 24px;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
