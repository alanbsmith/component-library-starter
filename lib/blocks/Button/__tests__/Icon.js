import React from 'react';
import Icon from '../Icon';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';

describe('Icon Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Icon name="home" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the left modifier', () => {
    const tree = renderWithTheme(
      <Icon
        modifiers={['left']}
        name="home"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the right modifier', () => {
    const tree = renderWithTheme(
      <Icon
        modifiers={['right']}
        name="home"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
