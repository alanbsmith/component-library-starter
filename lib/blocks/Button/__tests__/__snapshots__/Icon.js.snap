// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Element adds the left modifier 1`] = `
.c0 {
  font-size: 24px;
  padding-right: 0.5rem;
}

.c1 {
  color: #424242;
  font-size: 1.25rem;
}

<span
  aria-hidden={true}
  className="fa fa-home c0 c1"
  modifiers={
    Array [
      "left",
    ]
  }
/>
`;

exports[`Icon Element adds the right modifier 1`] = `
.c0 {
  font-size: 24px;
  padding-left: 0.5rem;
}

.c1 {
  color: #424242;
  font-size: 1.25rem;
}

<span
  aria-hidden={true}
  className="fa fa-home c0 c1"
  modifiers={
    Array [
      "right",
    ]
  }
/>
`;

exports[`Icon Element renders correctly 1`] = `
.c0 {
  font-size: 24px;
}

.c1 {
  color: #424242;
  font-size: 1.25rem;
}

<span
  aria-hidden={true}
  className="fa fa-home c0 c1"
/>
`;
